deployments/gpu-operator/charts/node-feature-discovery/values.yaml

```diff
@@ -91,17 +97,17 @@ master:
     type: ClusterIP
     port: 8080
 
-  resources: {}
-    # We usually recommend not to specify default resources and to leave this as a conscious
-    # choice for the user. This also increases chances charts run on environments with little
-    # resources, such as Minikube. If you do want to specify resources, uncomment the following
-    # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
-    # limits:
-    #   cpu: 100m
-    #   memory: 128Mi
-    # requests:
-    #   cpu: 100m
-    #   memory: 128Mi
+  resources:
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      # You may want to use the same value for `requests.memory` and `limits.memory`. The “requests” value affects scheduling to accommodate pods on nodes.
+      # If there is a large difference between “requests” and “limits” and nodes experience memory pressure, the kernel may invoke
+      # the OOM Killer, even if the memory does not exceed the “limits” threshold. This can cause unexpected pod evictions. Memory
+      # cannot be compressed and once allocated to a pod, it can only be reclaimed by killing the pod.
+      # Natan Yellin 22/09/2022 https://home.robusta.dev/blog/kubernetes-memory-limit
```

Comment:
This node and the default limits doesn't make sense. Limit is set at 4Gi, while request is 128Mi. Does that mean by default evictions are likely under memory pressure?

Reply:
@ArangoGutierrez

Reply:
As far as I am aware, when you have identical requests and limits, the Kubernetes scheduler prioritises the pod at schedule-time. This may be a good reason to encourage that practice and have defaults that reflect this. 
However,  It is difficult to provide sane defaults here as cluster size varies greatly in the user base. 

I think it would be consistent for us to provide default requests and limits that are consistent with the operator resource requests and limits. Perhaps, we can include those defaults in the gpu-operator chart values.yaml or just copy over the resource constraints found in NFD chart values over to the gpu-operator chart values

Reply:
The same basis with which we provided the gpu-operator resource constraints, we use to determine the default resource constraints of NFD:

```
  resources:
    limits:
      cpu: 500m
      memory: 350Mi
    requests:
      cpu: 200m
      memory: 100Mi

```

Reply:
is worth mention that this is just for `nfd-master` , `nfd-worker` is:
```yaml
  resources:
    limits:
      memory: 60Mi
    requests:
      cpu: 50m
      memory: 40Mi
```

Reply:
if we can setup priorityclassname for this, then the eviction problem can be avoided. Lets merge this and add a change to set priorityclass as "system-node-critical"